src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -592,7 +593,7 @@ private bool ValidatePackagesSha512(PackagesLockFile lockFile, LockFile assetsFi
                             var libraries = target.Dependencies
                                 .Where(dep => dep.Type != PackageDependencyType.Project)
                                 .Select(dep => new LibraryIdentity(dep.Id, dep.ResolvedVersion, LibraryType.Package))
-                                .ToList();
+                                .AsList();
```

Comment:
This change is unnecessary. There's no functional change here. 
LINQ generates enumerates which had to be converted to List.